### PR TITLE
Pipe painter

### PR DESCRIPTION
**Pull request by lzk228**
## About the PR

taken from https://github.com/SerbiaStrong-220/space-station-14/pull/73

**Media**

https://user-images.githubusercontent.com/16546901/245219430-0735db52-9f77-4086-91e1-6f3745543ca4.mp4
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: 
- add: Added pipe painter

### CODE REVIEW

**Comment by BRINGit34** [comment]
This is so rad

**Comment by potato1234x** [comment]
ok but what if it could paint everything 😳

**Review by ficcialfaint — CHANGES_REQUESTED** [review]
What’s the point of just taking someone’s work from another fork, not ensuring it works, not running it on the local server, not trying to fix any issues?

**Comment by c4llv07e** [comment]
> What’s the point of just taking someone’s work from another fork, not ensuring it works, not running it on the local server, not trying to fix any issues?

I think it was working two weeks ago and it just broke after BoundUserInterface refactoring.

File: Content.Client/PipePainter/PipePainterBoundUserInterface.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+using Content.Shared.PipePainter;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.PipePainter;
+
+public sealed class PipePainterBoundUserInterface : BoundUserInterface
+{
+    private PipePainterWindow? _window;
+
+    public PipePainterBoundUserInterface(ClientUserInterfaceComponent owner, Enum uiKey) : base(owner, uiKey)
```
Comment (c4llv07e) [inline Content.Client/PipePainter/PipePainterBoundUserInterface.cs:11]:
```suggestion
    public PipePainterBoundUserInterface(EntityUid owner, Enum uiKey) : base(owner, uiKey)
```

Now EntityUid is required as the owner.

File: Content.Server/PipePainter/PipePainterSystem.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    /// <summary>
+    /// Do on activation - open UI
+    /// </summary>
+    private void OnActivate(EntityUid uid, PipePainterComponent component, ActivateInWorldEvent args)
+    {
+        if (!EntityManager.TryGetComponent(args.User, out ActorComponent? actor))
+            return;
+
+        UpdateUiState(uid, component);
+        component.Owner.GetUIOrNull(PipePainterUiKey.Key)?.Open(actor.PlayerSession);
```
Comment (c4llv07e) [inline Content.Server/PipePainter/PipePainterSystem.cs:81]:
```suggestion
        _userInterfaceSystem.TryOpen(uid, PipePainterUiKey.Key, actor.PlayerSession);
```

This should work (I tested).

**Comment by lzk228** [comment]
works right

**Comment by Vaaankas** [comment]
Shouldnt it be the same item with the airlock painter?

**Comment by potato1234x** [comment]
> Shouldnt it be the same item with the airlock painter?

merge them into the omnipainter

**Review by AJCM-git — CHANGES_REQUESTED** [review]
> Shouldnt it be the same item with the airlock painter?

yeah they should

**Comment by lzk228** [comment]
bruh

**Comment by ficcialfaint** [comment]
> > What’s the point of just taking someone’s work from another fork, not ensuring it works, not running it on the local server, not trying to fix any issues?
> 
> 
> 
> I think it was working two weeks ago and it just broke after BoundUserInterface refactoring.

anyways, it'd never happen if they at least ensured it worked and copied the latest version from that codebase, had to test it locally first

**Comment by AJCM-git** [comment]
@lzk228 are you coming back to this?

**Comment by lzk228** [comment]
nah bruh

**Pull request closed without merging**